pipelime/commands/general.py

```diff
@@ -951,3 +951,60 @@ def _get_hash_key(self, keys: t.Sequence[str]) -> str:
             if key not in keys:
                 return key
             key += "_"
+
+
+class CopySharedItemsCommand(PipelimeCommand, title="copy-shared-items"):
+    """Copy shared items from a source dataset to a destination dataset. Datasets may
+    not have the same length."""
+
+    source: pl_interfaces.InputDatasetInterface = (
+        pl_interfaces.InputDatasetInterface.pyd_field(
+            alias="src",
+            piper_port=PiperPortType.INPUT,
+            description=(
+                "Where the shared items are copied from. Must have at least one "
+                "sample and one shared item."
+            ),
+        )
+    )
+    dest: pl_interfaces.InputDatasetInterface = (
+        pl_interfaces.InputDatasetInterface.pyd_field(
+            alias="dst",
+            piper_port=PiperPortType.INPUT,
+            description=(
+                "Where the shared items are copied to. Must have at least one sample "
+                "and any number of shared items. Source and destination datasets may "
+                "not have the same length, as only the shared items are copied."
+            ),
+        )
+    )
+    output: pl_interfaces.OutputDatasetInterface = (
+        pl_interfaces.OutputDatasetInterface.pyd_field(
+            alias="o",
+            piper_port=PiperPortType.OUTPUT,
+            description=("Where the resulting dataset is written to."),
+        )
+    )
+
+    key_list: t.Sequence[str] = pyd.Field(
+        ...,
+        alias="k",
+        description=("The keys to copy. Must be present in source dataset."),
+    )
+
```

Comment:
Forse aggiungerei un flag per forzare la copia degli item shared, altrimenti ogni sample si becca un hardlink, che in generale va bene, ma in certi casi potrebbe non essere quello che l'utente vuole. È sufficiente poi impostare la property `serialization_mode` direttamente sui singoli `Item`.